typescript-getting-started/README.md

```diff
@@ -0,0 +1,69 @@
+## Cloud Functions for Firebase: getting started with TypeScript
+
+This is an example Firebase project for using
+[TypeScript](https://www.typescriptlang.org/) with
+[Cloud Functions for Firebase](https://firebase.google.com/products/functions)
+
+### Why TypeScript?
+
+[TypeScript](https://www.typescriptlang.org/) is a typed superset of JavaScript
+that compiles to plain JavaScript.
+
+One of the biggest challenges with developing in JavaScript is that it is
+very easy to write code that has runtime errors. TypeScript enables the fast
+development of JavaScript with optional types. When types are used,
+supported editors provide auto-suggest for methods and properties along
+with syntax highlighting of errors, which speeds development.
+
+TypeScript supports targeting different browsers, and optimizes
+the resulting JavaScript. It is much easier to write clean, consistent code
+across a project and development team.  TypeScript offers support for the
+latest and evolving JavaScript features like async functions and decorators,
+to help build robust components.
+
+If you are new to TypeScript, check out the [TypeScript PlayGround](https://www.typescriptlang.org/play/index.html).
+
+
+### Project Setup
+
+This project was set up using [yarn](https://yarnpkg.com) which will ensure
+that your deployed Cloud Functions are using the exact same versions of
+npm modules that you are running locally. The specific version of each
+dependency is saved in [yarn.lock](functions/yarn.lock)
+
+
+This example has the default sample function. To use this example as a
+starter project:
+
+1. `npm install -g firebase-tools`
+2. `git clone https://github.com/firebase/functions-samples.git`
```

Comment:
oof. I'm not sure if this is clearer or not, but I was able to tinker and find this:

```bash
curl https://codeload.github.com/firebase/functions-samples/tar.gz/master |  \
  tar -xz --strip=2 functions-samples-*/typescript-getting-started
```

This has the benefit of not creating a git repo that may not be compatible with any existing repros and gets rid of the need for the `rm -rf` & `cp -r` lines below. Don't really have an opinion which is better; just thought I'd give you an FYI.

Reply:
Nice suggestion.  I agree it is more verbose as is, but I think it requires less knowledge of bash, so will leave it.